### ISSUE TITLE
quality of ol.layer.Tile is not as good as ol.layer.Image

### ISSUE THREAD

**Issue by danzaharescu** [openlayers/openlayers#8323]
Hello,

I have identified a new possible problem. While playing with types described in title, I have encountered subtle differences in rendering quality.
The tile object suddenly deteriorates at a certain zoom out, and in addition, pixelRatio: 2 parameter added in the map does not have any effect.
On the other hand, ol.layer.Image is very clean and smooth at any zoom (especially, when pixelRatio: 2 is used)

I am attaching my test scenario.
You may find my  'let useTile = false' statement, and change it to true, to use tiles instead of image.
When doing a few (4-5) zoom outs in the html page, for the time use case the sudden quality decrese is visible.

I have checked the map, view and tile object specification, but I was not able to find something helpful yet.

Regards,
Dan

**Comment by danzaharescu** [comment]
[test_ol.zip](https://github.com/openlayers/openlayers/files/2141519/test_ol.zip)
Forgot the attachement... you may need to adapt  the line:
let myurl = 'c:\\sources\\test_text.png';
from the html file.

**Comment by ahocevar** [comment]
I think you are referring to #8316. Setting a `pixelRatio` on the map is not something I would recommend for production, it is just something I had mentioned for demonstration purposes.

Regarding this ticket, pixel ratio for tile sources is controlled using the `tilePixelRatio` option on the source, but requires tiles with a higher resolution. It does not make much sense to render a static image as tile. You need a tile server to serve tiles.

I'm closing this ticket, because zooming out for tile layers means that new tiles will be loaded, which are optimized for the new zoom level.

**Comment by danzaharescu** [comment]
Hello Ahocevar,,

Thanks for the interest in the problem.
But, even if the ticket is already closed, please let me post some additional considerations.
I have some expectations, and some of them may be legitimate, some not.
You and the team could analyze and decide upon these:

1. ol.layer.Tile  and ol.layer.Image should somehow behave similar at the same zooms.
   But there seem to be some differences in rendering, as I stated in title.  If my understanding is correct,  
   the only differences between these objects should regard centering/positioning/extent (example: tile 
   requires tile grid, image not). Uniform rendering algorithms across objects could make life of 
   developer easier I think...
 
2. Strictly about the current ticket,  zooming out should, in my view, definitely not lead to a blurring of 
   the image (maybe I should have made the statement more clear from beginning), but I would expect a 
   gradual sharpening of the image (the effect that is corrected by pixelRatio ) . 
   Let's say, my image is clear at 105%, but blurry at 100% ...I think this is not normal. I like how 
   ol.layer.Image behaves in these zoom ranges (however, at about 125% and more, ol.layer.Image    
   becomes slightly blurrier than Tile unfortunately).

   You say I should bring other tiles appropriate for this zoom level (let's say 100%). But if I bring other 
   lower resolution tiles, the image will be even blurrier.  And bringing higher resolution tiles does not 
   make sense,  because my image already has enough quality for this zoom (as I said, it looks good at 
   105% ++   zooms, and ol.layer.Image also proves this).
 
   So my conclusion is, either is a coding problem from my side (missing some parameters, or something)
   either I did not describe the problem in the ticket clear enough.
   Maybe the problem was somehow missed so far, because OpenLayers is usually used for drawing 
   maps, but  in my test case there is a picture containing text, which could make the issue more obvious.

It's up to you, if you will reconsider double check this issue or not.
I am willing to further contribute to fixing such issues (provide details, screenshots, test code, test possible snapshot/internal fixes from your side), if there is also an interest from your side.

My regards,
Dan

**Comment by danzaharescu** [comment]
Hello Ahocevar,

I have tried tilePixelRatio: 2 in the source, and the quality is now improved!
My mistake was that I tried to set it in the olmap as for the ImageType (seemed to fit in the same place, and I was not careful enough to your post, apparently).
Still, the image seems to be less 'smooth' compared with ol.Layer.Image, I can only think that inside there are different algorithms.
At the moment, I would say ol.Layer.Image behaves better in most cases, except when the zoom becomes larger. 
I think some investigation could be done in the future, to find a way to offer a common behavior and quality (let's say, best of both worlds -best meaning in my opinion as Image at lower zooms, and as Tile at higher zooms)

Thanks a lot and regards,
Dan
